src/main/java/org/opentripplanner/routing/algorithm/raptor/transit/TripScheduleWrapperImpl.java

```diff
@@ -0,0 +1,51 @@
+package org.opentripplanner.routing.algorithm.raptor.transit;
+
+import org.opentripplanner.routing.edgetype.TripPattern;
+import org.opentripplanner.routing.trippattern.TripTimes;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class TripScheduleWrapperImpl implements TripSchedule {
+
+    private static final Logger LOG = LoggerFactory.getLogger(TripScheduleWrapperImpl.class);
+
+    private final TripTimes originalTripTimes;
+
+    private final TripPattern originalTripPattern;
+
+    public TripScheduleWrapperImpl (TripTimes originalTripTimes, TripPattern originalTripPattern) {
+        this.originalTripTimes = originalTripTimes;
+        this.originalTripPattern = originalTripPattern;
+    }
+
+    @Override
+    public TripTimes getOriginalTripTimes() {
+        return originalTripTimes;
+    }
+
+    @Override
+    public TripPattern getOriginalTripPattern() {
+        return originalTripPattern;
+    }
+
+    @Override
+    public int getServiceCode() {
+        return originalTripTimes.serviceCode;
+    }
+
+    @Override
+    public int arrival(int stopPosInPattern) {
+        return originalTripTimes.getArrivalTime(stopPosInPattern);
+    }
+
+    @Override
+    public int departure(int stopPosInPattern) {
+        return originalTripTimes.getDepartureTime(stopPosInPattern);
+    }
+
+    @Override
+    public String debugInfo() {
+        throw new UnsupportedOperationException();
+    }
+
+}
```

Comment:
In this case it wasn't really about being flexible, it was about showing two working possibilities: a wrapper around the existing OTP1 objects implementing the Raptor interface, versus TripScheduleImpl which copies the references to the internal arrivals and departures arrays to avoid one layer of indirection. I assumed we would choose one or the other once we had some evidence on efficiency.

The third and maybe best option is to simply make the existing OTP1 objects implement the OTP2 interface and use them directly.

Reply:
OK, I need to get that QA Tool done - so we can begin tracing performance over time and easily test things like this. I am ok with leaving both classes in there and we can test it later.

Reply:
Added TODO